ci/clean_notebooks.py

```diff
@@ -35,44 +62,56 @@ def clean_notebook(file: pathlib.Path, check_only=False) -> None:
     if check_only:
         print(f"Checking {file}")
 
-    # Remove the output and metadata from each cell
-    # also reset the execution count
-    # if the cell has no code, remove it
     for cell in nb.cells:
-        if "outputs" in cell and cell["outputs"]:
-            if check_only:
-                raise UncleanNotebookError(f"Notebook {file} has outputs")
-            cell["outputs"] = []
-            was_dirty = True
-        if "metadata" in cell and cell["metadata"]:
-            if check_only:
-                raise UncleanNotebookError(f"Notebook {file} has metadata")
-            cell["metadata"] = {}
-            was_dirty = True
-        if "execution_count" in cell and cell["execution_count"]:
-            if check_only:
-                raise UncleanNotebookError(f"Notebook {file} has execution count")
-            cell["execution_count"] = None
-            was_dirty = True
+
+        # Remove empty cells
         if cell["cell_type"] == "code" and not cell["source"]:
             if check_only:
                 raise UncleanNotebookError(f"Notebook {file} has empty code cell")
             nb.cells.remove(cell)
             was_dirty = True
 
+        # Clean the cell
+        # (copy the cell keys list so we can iterate over it while modifying it)
+        for key in list(cell):
+            if key not in structure[cell["cell_type"]]:
+                if check_only:
+                    raise UncleanNotebookError(
+                        f"Notebook {file} has unknown cell key {key}",
+                    )
+                del cell[key]
+                was_dirty = True
+            else:
+                cell_structure = structure[cell["cell_type"]][key]
+                if cell_structure["do"] == "keep":
+                    continue
+                elif cell_structure["do"] == "default":
+                    default_value = cell_structure["value"]
+                    if cell[key] != default_value:
```

Comment:
`"default"` feels like a bit of a misnomer here given you're forcing it to always have that value. That's not a default value, that's the only legal value. Perhaps rename it to something like `"constant"`?